network/crl.go

```diff
@@ -53,6 +58,59 @@ func NewCRLConfig(uri, fromCert string) (*CRLConfig, error) {
 	return &CRLConfig{uri: uri, fromCert: fromCertVal}, nil
 }
 
+type CRLClient interface {
+	// Fetch fetches and parses CRL from passed URI (can be either http:// or file://)
+	Fetch(uri string) (*pkix.CertificateList, error)
+}
```

Comment:
One can argue that it might be better to have the interface like this:

```go
type CRLClient interface {
        FetchCRL() (*pkix.CertificateList, error)
}
```
where the client is configured to always fetch from a particular location (and might be reconfigured later, if needed). This separates sources (local files vs. networks) which can be just different implementations, if necessary, without a giant switch in the current one.

Such interface also does not expose the (arguably) unnecessary coupling of CRLs to URIs. Right now the interface insists that a CRL *must* be identified with a URI. This might not be the case, for example, with test stubs, or maybe some fixed CRL hardwired into Acra.

Reply:
It's okay if we rely only on static configured crl URIs. But we have one more source of URI: client's certificate. And when we accept new connection from client and he passes certificate with [crlDistributionPoint](https://tools.ietf.org/html/rfc5280#section-4.2.1.13), then we should use some client to fetch CRL data.
I discovered from rfc that crl URI may be not only HTTP URL, it may be ldap access point too, and probably other protocols for CRL fetching.